scripts/release.sh

```diff
@@ -183,7 +183,7 @@ main() {
     fi
 
     # Tag release.
-    if [ "$(git tag --list | grep -c "${RELEASE_VERSION}")" -gt 0 ]; then
+    if git tag --list | grep --quiet "^${RELEASE_VERSION}$"; then
```

Comment:
After second thought, I think it might be better to keep it as it's for the stable releases, because they will never run into the `v3.6.0 < v3.6.0-rc.x` problem. @ivanvc @jmhbnz wdyt?

Please also see https://github.com/etcd-io/etcd/pull/19955#discussion_r2092449883

Reply:
Hmm I would rather keep our release script aligned so we have consistent behaviour across all the branches we release.

Agree I don't expect we would release a candidate or any kind of variant tag for a patch but I would still prefer aligning the script behaviour across all the branches so we know what to expect when we run it.

Reply:
OK, let's merge it.